python/ray/serve/_private/http_proxy.py

```diff
@@ -615,6 +616,32 @@ def __init__(
         if http_middlewares is None:
             http_middlewares = []
 
+        if RAY_SERVE_HTTP_PROXY_CALLBACK_IMPORT_PATH:
+            middlewares = call_function_from_import_path(
+                RAY_SERVE_HTTP_PROXY_CALLBACK_IMPORT_PATH
+            )
+            if middlewares is None:
+                middlewares = []
+            if not isinstance(middlewares, list):
+                raise ValueError(
+                    f"HTTP proxy callback {RAY_SERVE_HTTP_PROXY_CALLBACK_IMPORT_PATH} "
+                    "must return a list of Starlette middlewares."
+                )
+            else:
+                # All middlewares must be Starlette middlewares.
+                # https://www.starlette.io/middleware/#using-pure-asgi-middleware
+                for middleware in middlewares:
+                    if not issubclass(
+                        type(middleware), starlette.middleware.Middleware
+                    ):
+                        raise ValueError(
+                            "HTTP proxy callback "
+                            f"{RAY_SERVE_HTTP_PROXY_CALLBACK_IMPORT_PATH} "
+                            f"must return a list of Starlette middlewares, "
+                            f"instead got {type(middleware)} type item in the list."
+                        )
```

Comment:
would be nice to put this in its own method/function to improve readability
```suggestion
            middlewares = _validate_callback_return(call_function_from_import_path(
                RAY_SERVE_HTTP_PROXY_CALLBACK_IMPORT_PATH
            ))
```